lib/watchful.ts

```diff
@@ -15,20 +15,25 @@ import { WatchApiGatewayOptions, WatchApiGateway } from './api-gateway';
 
 export interface WatchfulProps {
   readonly alarmEmail?: string;
-  readonly alarmSqs?: string;
+  readonly alarmSqs?: sqs.IQueue;
+  readonly alarmSns?: sns.ITopic;
 }
 
 export class Watchful extends Construct implements IWatchful {
   private readonly dash: cloudwatch.Dashboard;
-  private readonly alarmTopic?: sns.Topic;
+  private readonly alarmTopic?: sns.ITopic;
 
   constructor(scope: Construct, id: string, props: WatchfulProps = { }) {
     super(scope, id);
 
-    if (props.alarmEmail || props.alarmSqs) {
+    if ((props.alarmEmail || props.alarmSqs) && !props.alarmSns) {
       this.alarmTopic = new sns.Topic(this, 'AlarmTopic', { displayName: 'Watchful Alarms' });
     }
 
+    if (props.alarmSns) {
```

Comment:
I think it still is because otherwise it would override the `alarmTopic` it creates above and `props.alarmSns` could be `undefined` coming in so then `this.alarmTopic` would be set to undefined instead of the one that gets created if they pass in `props.alarmEmail || props.alarmSqs`

Maybe I'm not seeing it correctly though. Could you explain further?